### ISSUE TITLE
TableName assumes PUBLIC schema when no schema is set

### ISSUE THREAD

**Issue by emmanueljob** [databricks/spark-redshift#295]
https://github.com/databricks/spark-redshift/blob/master/src/main/scala/com/databricks/spark/redshift/TableName.scala#L42

Shouldn't this follow the rules of Redshift and respect the search_path of the logged in user:

http://docs.aws.amazon.com/redshift/latest/dg/r_search_path.html

**Comment by JoshRosen** [comment]
Good catch. If I'm reading the linked documentation correctly, it sounds like the right fix is to pass through non-schema-qualified table names `as-is` without assuming `PUBLIC`.

Which version of `spark-redshift` are you using / against which releases should I target a patch?

**Comment by JoshRosen** [comment]
Ping @emmanueljob, would be great to know in which maintenance branches you would like this fixed.

**Comment by emmanueljob** [comment]
I have no specific urgent need, we can work around the issue for now. If it's in whatever your next release is whenever that is we'll take advantage of it.

**Comment by sboora** [comment]
@JoshRosen any chance this can be patched up in v2.0.1, a couple of our ETL pipelines are failing.

**Comment by sboora** [comment]
I am taking a stab at this here, 
https://github.com/databricks/spark-redshift/compare/branch-2.x...sboora:branch-2.x

**Comment by venkatvb** [comment]
We're also running into the same bug, working around it by explicitly specifying the schema name. It'll be reasonable to default to `search_path` if no schema is set.
